python/ray/serve/scripts.py

```diff
@@ -441,3 +476,49 @@ def delete(address: str, yes: bool):
     cli_logger.newline()
     cli_logger.success("\nSent delete request successfully!\n")
     cli_logger.newline()
+
+
+@cli.command(
+    short_help="Build a Serve app into a structured config.",
+    help=(
+        "Imports the Serve app at IMPORT_PATH and generates a structured "
+        "config for it that can be used by `serve deploy` or the REST API. "
+    ),
+)
+@click.option(
+    "--app-dir",
+    "-d",
+    default=".",
+    type=str,
+    help=APP_DIR_HELP_STR,
+)
+@click.option(
+    "--output-path",
+    "-o",
+    default=None,
+    type=str,
+    help=(
+        "Local path where the output config will be written in YAML format. "
+        "If not provided, the config will be printed to STDOUT."
+    ),
+)
+@click.argument("import_path")
+def build(app_dir: str, output_path: Optional[str], import_path: str):
+    sys.path.insert(0, app_dir)
+
+    node: DeploymentNode = import_attr(import_path)
+    if not isinstance(node, DeploymentNode):
+        raise TypeError(
+            f"Expected '{import_path}' to be DeploymentNode, but got {type(node)}."
+        )
+
+    app = serve.build(node)
+
+    if output_path is not None:
+        if not output_path.endswith(".yaml"):
+            raise ValueError("FILE_PATH must end with '.yaml'.")
+
+        with open(output_path, "w") as f:
```

Comment:
Change to mode to `"w+"` since the config file might not exist yet. Also, should we log a warning or ask confirmation when overwriting an existing file?

Reply:
the `+` is just for read and write, `w` already does what we want